src/main/java/com/box/sdk/BoxJSONResponse.java

```diff
@@ -31,33 +34,60 @@ public BoxJSONResponse(HttpURLConnection connection) {
     }
 
     /**
+     * Constructs a BoxAPIResponse with an http response code and response body.
+     * @param responseCode http response code
+     * @param httpHeaders map of http headers
+     * @param body response body as Json Object
+     */
+    public BoxJSONResponse(int responseCode, Map<String, String> httpHeaders, JsonObject body) {
+        super(responseCode, httpHeaders);
+        this.jsonObject = body;
+        this.json = body.toString();
+    }
+
+    /**
+     * Get response as Json Object.
+     * @return response as JsonObject
+     */
+    public JsonObject getJsonObject() {
+        if (this.jsonObject != null) {
+            return this.jsonObject;
+        } else {
+            return JsonObject.readFrom(this.getJSON());
+        }
+    }
+
+    /**
      * Gets the body of the response as a JSON string. When this method is called, the response's body will be read and
      * the response will be disconnected, meaning that the stream returned by {@link #getBody} can no longer be used.
      * @return the body of the response as a JSON string.
      */
     public String getJSON() {
         if (this.json != null) {
             return this.json;
-        }
+        } else if (this.jsonObject != null) {
+            return this.jsonObject.toString();
+        } else {
 
-        InputStreamReader reader = new InputStreamReader(this.getBody(), StandardCharsets.UTF_8);
-        StringBuilder builder = new StringBuilder();
-        char[] buffer = new char[BUFFER_SIZE];
+            InputStreamReader reader = new InputStreamReader(this.getBody(), StandardCharsets.UTF_8);
```

Comment:
space / tab

Reply:
that's correct. I have this enclosed  in an else block now